include/swift/Option/Options.td

```diff
@@ -368,6 +368,10 @@ def emit_objc_header_path : Separate<["-"], "emit-objc-header-path">,
          ArgumentIsPath]>,
   MetaVarName<"<path>">, HelpText<"Emit an Objective-C header file to <path>">;
 
+def emit_archive : Flag<["-"], "emit-archive">,
+  Flags<[FrontendOption, NoInteractiveOption, DoesNotAffectIncrementalBuild]>,
```

Comment:
I'll leave it to @jrose-apple to decide if this is how this should happen, but if this is going to affect the module, then it should go in FrontendOptions and have the `ModuleInterfaceOption` flag added, so it is included in the set of command-line flags that go into a `.swiftinterface` file.

Reply:
It's a driver flag as well as a frontend flag, so this is the correct file, but Harlan is correct that it is a `ModuleInterfaceOption`, and it certainly does affect the incremental build.

Reply:
If `ModuleInterfaceOption` ensures it's automatically serialised, does that mean the `IsStaticLibrary` flag I've added elsewhere is redundant?

Reply:
More specifically: I see that `CompilerInstance::getMainModule` copies the frontend arguments to the main module; does that also apply to deserialised modules or are the `options_block::IS_STATIC_LIBRARY` and `serialization::ExtendedValidationInfo` changes necessary?